shuttle/db/db_functions.py

```diff
@@ -111,17 +111,22 @@ def username_search(username):
         return abort(503)
 
 
+# Commits schedule to the db. Assumes first row has the locations with its times in the same column
 def commit_schedule(table, all_locations):
     try:
         all_locations = [i.upper() for i in all_locations]
         queries = []
+        locations = {}
         for i in range(len(table)):
-            location = ''
             for j in range(len(table[i])):
                 departure_time = 0
-                if table[i][j].upper() in all_locations:
-                    location = table[i][j]
+                if i == 0:
+                    if table[i][j].upper() in all_locations:
+                        locations.update({j: table[i][j]})
+                    else:
+                        return 'no match'
                 elif table[i][j] == '-' or table[i][j] == 'DROP':
+                    # August will be used to tell if it has no time
```

Comment:
I am confused by this, could you explain more as a reply to this comment?

Reply:
I set each type of information (location, time, "-") as a certain date so I would be able to tell what type of data it is in the database. I guess there might be a better way to do it but this seems to work in the meantime. 
So if the information in the database was "-", I would set its time would have a August date. Then when I take that information from the database, the time tells me what kind of data it is (since it would have a location but no way to tell that it isn't a time or just a location). Hopefully that makes sense

Reply:
I updated the database so the default value for departure_time is NULL so now you can just not put a departure_time in and whenever departure_time is NULL you know it should be '-'

Reply:
> I set each type of information (location, time, "-") as a certain date

Why would we need a db record with just a location? We should be able to get all the locations by just iterating through the table and inserting locations into a dictionary or a similar data structure.

Reply:
Maybe I'm missing the obvious but how would we be able to tell just by looking at the db which locations span the first row? Since a few locations repeat, it doesn't seem obvious which ones to use from a computer's perspective

Reply:
What do you mean by span the first row? I assume that the order that you insert them into the db could be an indication on how they should be ordered in the table. Anyhow, we should probably find a way to do this that doesn't have to use specific dates to figure out what goes where.

Reply:
Well the first row of the schedule just holds locations. I could insert the locations first but if the time is null, there isn't an obvious way to me to tell what is a location and what is a "-". And since locations repeat, it's hard to find a pattern in the first row

Reply:
I guess I could maybe use the initial spreadsheet to figure that out